pkg/intercp/client/pool_test.go

```diff
@@ -0,0 +1,99 @@
+package client_test
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/connectivity"
+
+	"github.com/kumahq/kuma/pkg/intercp/client"
+)
+
+var _ = Describe("Pool", func() {
+	It("should not create a client when TLS is not configured", func() {
+		// given
+		pool := client.NewPool(nil, 1*time.Second)
+
+		// when
+		_, err := pool.Client("http://192.168.0.1")
+
+		// then
+		Expect(err).To(Equal(client.TLSNotConfigured))
+	})
+
+	Context("configured with TLS", func() {
+		var pool *client.Pool
+		const idleDeadline = 100 * time.Millisecond
+
+		BeforeEach(func() {
+			pool = client.NewPool(func(s string, config *client.TLSConfig) (client.Conn, error) {
+				return &testConn{
+					state: connectivity.Ready,
+				}, nil
+			}, idleDeadline)
+			pool.SetTLSConfig(&client.TLSConfig{})
+			go pool.StartCleanup(context.Background(), 10*time.Millisecond)
+		})
+
+		It("should keep the connection open", func() {
+			// when
+			c, err := pool.Client("http://192.168.0.1")
+			Expect(err).ToNot(HaveOccurred())
+			c2, err := pool.Client("http://192.168.0.1")
+			Expect(err).ToNot(HaveOccurred())
+
+			// then
+			Expect(c).To(BeIdenticalTo(c2))
+		})
+
+		It("should create a new connection after the previous expired", func() {
+			// given
+			c, err := pool.Client("http://192.168.0.1")
+			Expect(err).ToNot(HaveOccurred())
+
+			// when
+			time.Sleep(idleDeadline * 2)
```

Comment:
nitpick: shouldn't use real sleep in tests

Reply:
fixed